File: src/spice2x/api/resources/dart/spiceapi/src/wrappers/coin.dart
part of spiceapi;

Future<int> coinGet(Connection con) {
  var req = Request("coin", "get");
  return con.request(req).then((res) {
    return res.getData()[0];
  });
}

Future<void> coinSet(Connection con, int amount) {
  var req = Request("coin", "set");
  req.addParam(amount);
  return con.request(req);
}

Future<void> coinInsert(Connection con, [int amount=1]) {
  var req = Request("coin", "insert");
  if (amount != 1)
    req.addParam(amount);
  return con.request(req);
}

File: src/spice2x/api/resources/python/spiceapi/coin.py
from .connection import Connection
from .request import Request


def coin_get(con: Connection):
    res = con.request(Request("coin", "get"))
    return res.get_data()[0]


def coin_set(con: Connection, amount: int):
    req = Request("coin", "set")
    req.add_param(amount)
    con.request(req)


def coin_insert(con: Connection, amount=1):
    req = Request("coin", "insert")
    if amount != 1:
        req.add_param(amount)
    con.request(req)
